### PR TITLE
Feature 643 provide pds streams per rest

### LINKED ISSUE #643: Provide error and system out data inside PDSJob data in database + Rest access
# Usecases

## Admin wants to get error stream data per REST API for PDS job
As an administrator I can fetch the error stream output for a PDS job which has been run.
The admin knows the PDS job UUID

## Admin wants to get output stream data per REST API for PDS job
As an administrator I can fetch the out stream output for a PDS job which has been run.
The admin knows the PDS job UUID

# Technical details
- We need to add two additional text fields to PDSJob entity
- We do not change the existing admin rest call which can fetch result or error. this will be done in another issue!

### LINKED ISSUE #801: Integration tests check interval in adapters have wrong configured values in profile
## Situation
Inspecting a broken build from #643 I found:

```
:06:55.247  WARN 9008 --- [93e391f904a6-24] c.d.s.a.AbstractAdapterConfigBuilder     : Configured check interval:-694967296 milliseconds is lower than minimum value:500 milliseconds. Automatic reset to default value: 60000 milliseconds. Please check your configuration!

```
This is very sad, because the check intervall is set to 60000 - maybe this slows down tests!

### Reproduction
1. Start a integration test server local in your IDE
2. Execute CancelJobScenario4IntTest#cancel_a_long_running_codescan_job
3. Inspect the server log output in console for "Configured check interval"

## Wanted
1. This shall never happen in the integration tests
2. The log should show up the builder class, so it's clear where this happend!

## Solution
1. Change log output and show class information
2. Identify integration test profile problem and fix it, so that this no longer happens

### PR DESCRIPTION
**Pull request by de-jcup**
This PR
- closes #643
- closes #801

### CODE REVIEW

File: sechub-pds/src/test/java/com/daimler/sechub/pds/job/PDSUpdateJobStreamContentServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MIT
```
Comment (Jeeppler) [inline sechub-pds/src/test/java/com/daimler/sechub/pds/job/PDSUpdateJobStreamContentServiceTest.java:1]: Please apply formatter.

**Review by Jeeppler — CHANGES_REQUESTED** [review]
See comments.

Tested with PDS single and cluster instances.

File: sechub-doc/src/docs/asciidoc/diagrams/diagram_pds_overview_stream_access.puml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        pdsMember1->pdsDB: update **"last update timestamp"**
+    end    
+end
+pdsMember2->pdsDB: fetch stream content from DB field
+pdsMember2->pdsLoadBalancer: return content
+hnote across: End of current process still running case
+
+else job already finished
+pdsMember1->pdsDB:  write file to DB field after process ended 
+note over pdsMember2: Just fetch final data from DB".
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-doc/src/docs/asciidoc/diagrams/diagram_pds_overview_stream_access.puml:44]: done

File: sechub-doc/src/docs/asciidoc/diagrams/diagram_pds_overview_stream_access.puml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+pdsMember1->pdsJobProcess: starts process
+pdsJobProcess->pdsMember1: writes streams to filesystem
+note over pdsAdmin, pdsMember2: Administrator wants to fetch content stream\nbut loadbalancer uses other cluster member\nwhich does not execute the process".
+pdsAdmin->pdsLoadBalancer:Wants to fetch \nstream content
+pdsLoadBalancer->pdsMember2:delegates
+pdsMember2->pdsDB: check job finished
+
+alt job still running
+
+note over pdsMember2: Cluster member2 has no \ndirect access to the job\nwhich runs on another member!".
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-doc/src/docs/asciidoc/diagrams/diagram_pds_overview_stream_access.puml:23]: done

File: sechub-doc/src/docs/asciidoc/diagrams/diagram_pds_overview_stream_access.puml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+entity "PDS member 2" as pdsMember2
+entity "..." as pdsMemberOther
+entity "PDS member n" as pdsMemberN
+boundary "Job process" as pdsJobProcess
+database "PDS DB" as pdsDB
+
+pdsJobQueue<-pdsMember1: fetches Job and starts execution
+pdsMember1->pdsJobProcess: starts process
+pdsJobProcess->pdsMember1: writes streams to filesystem
+note over pdsAdmin, pdsMember2: Administrator wants to fetch content stream\nbut loadbalancer uses other cluster member\nwhich does not execute the process".
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-doc/src/docs/asciidoc/diagrams/diagram_pds_overview_stream_access.puml:16]: done

File: sechub-doc/src/docs/asciidoc/documents/shared/concepts/concept_pds_fetch_stream_of_running_job.adoc [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+The created process does not store the files on a shared volume or S3 bucket
+but on the local file system (like done for uploaded source files for example).
+
+So we can easily write those text content from the files to our database when 
+the OS process (executing PDS caller script) has finished or failed, because
+at this time we have direct access to the files.
+
+
+But when an administrator wants to get the current output for a running job
+/ operation this becomes tricky in clusted environment, because the process
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-doc/src/docs/asciidoc/documents/shared/concepts/concept_pds_fetch_stream_of_running_job.adoc:14]: done

File: sechub-doc/src/docs/asciidoc/documents/shared/concepts/concept_pds_fetch_stream_of_running_job.adoc [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+stream data is only available on the machine where the job is executed. 
+Using a loadbalancer we do not know which machine will be used for the REST 
+operation.
+
+So it becomes necessary, that the executing cluster member will store the
+error and output text data in database to provide this cluster wide.
+
+On the other hand it shall not slow down or block the execution machine because of
+permanent updates inside the DB. If an admin wants to check the output or 
+error stream content of a running job, the information shall be gathered lazily.
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-doc/src/docs/asciidoc/documents/shared/concepts/concept_pds_fetch_stream_of_running_job.adoc:24]: done

File: sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/AssertSecHubReport.java [hunk 7]
```diff
@@ -206,7 +207,15 @@ public AssertSecHubReport hasTrafficLight(TrafficLight trafficLight) {
             fail("No trafficlight found inside report!\nLast output line was:" + lastOutputLIne);
         }
         String trText = r.asText();
-        assertEquals(trafficLight, TrafficLight.fromString(trText));
+        TrafficLight foundTrafficLight = TrafficLight.fromString(trText);
+        if (!trafficLight.equals(foundTrafficLight)) {
+            /*
+             * in this case we log out the complete JSON content - interesting for debugging
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/AssertSecHubReport.java:213]: done

File: sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/ExecutionConstants.java [hunk 8]
```diff
@@ -3,39 +3,26 @@
...
-     * Id for default profile 1 - please NEVER CHANGE THIS PROFILE! Create your own execution
-     * profiles. This profile contains executors of CHECKMARX, NESSUS and NETSPARKER in version V1.
-     * Those executors do need their information by ENVIRONEMNT variables. This is a fix setup.
-     * PlEASE DO NOT CHAGNE IT! We use this profile to execute old tests to simulate old behaviour
-     * were no execution profiles nor executor configurations were available - all of those 
-     * scenarios (1-6) will have automatically this profile assigned to projects!  
+     * Id for default execution profile (integration test profile 1)) - please NEVER
+     * CHANGE THIS PROFILE! Create your own execution profiles. This profile
+     * contains executors of CHECKMARX, NESSUS and NETSPARKER in version V1. Those
+     * executors do need their information by ENVIRONEMNT variables. This is a fix
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/ExecutionConstants.java:19]: done

File: sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/TestAPI.java [hunk 9]
```diff
@@ -198,11 +200,24 @@ public boolean runImpl() throws Exception {
...
      */
-    @SuppressWarnings("unchecked")
     public static void waitForJobRunning(TestProject project, UUID jobUUID) {
-        LOG.debug("wait for job running project:{}, job:{}", project.getProjectId(), jobUUID);
+        waitForJobRunning(project, 5, 300, jobUUID);
+    }
 
-        TestAPI.executeUntilSuccessOrTimeout(new AbstractTestExecutable(SUPER_ADMIN, 5, HttpClientErrorException.class) {
+    /**
+     * Waits for sechub job being running
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/api/TestAPI.java:208]: done

File: sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/scenario2/Scenario2.java [hunk 10]
```diff
@@ -4,6 +4,7 @@
 import com.daimler.sechub.integrationtest.api.TestProject;
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-integrationtest/src/main/java/com/daimler/sechub/integrationtest/scenario2/Scenario2.java:4]: done

**Review by Jeeppler — APPROVED** [review]

**Pull request merged**